library/systemd/src/lib/yast2/system_service.rb

```diff
@@ -0,0 +1,490 @@
+    attr_reader :service
+
+    # @return [Hash<Symbol,Object>] Errors when trying to write changes to the
+    #   underlying system.
+    attr_reader :errors
+
+    # @return [Symbol, nil] :start, :stop, :restart or :reload. It returns nil
+    #   if no action has been requested yet.
+    attr_reader :action
+
+    # @!method support_reload?
+    #
+    # @return [Boolean]
+    def_delegator :@service, :can_reload?, :support_reload?
+
+    def_delegators :@service, :name, :static?, :running?, :description
+
+    class << self
+      # Finds a service by its name
+      #
+      # @param name [String] service name
+      # @return [SystemService, nil] nil if the service is not found
+      def find(name)
+        systemd_service = Yast::SystemdService.find(name)
+        return nil unless systemd_service
+
+        new(systemd_service)
+      end
+
+      # Finds a service by its name
+      #
+      # @param name [String] service name
+      #
+      # @raise [NotFoundError] if the service is not found
+      # @return [SystemService]
+      def find!(name)
+        system_service = find(name)
+        raise NotFoundError unless system_service
+
+        system_service
+      end
+
+      # Finds a set of services by their names
+      #
+      # @param names [Array<String>] service names to find
+      #
+      # @raise [NotFoundError] if any service is not found
+      # @return [Array<SystemService>]
+      def find_many(names)
+        systemd_services = Yast::SystemdService.find_many(names)
+        raise NotFoundError if systemd_services.any?(&:nil?)
+
+        systemd_services.map { |s| new(s) }
+      end
+    end
+
+    # Constructor
+    #
+    # @param service [Yast::SystemdServiceClass::Service]
+    def initialize(service)
+      @service = service
+      @changes = {}
+      @errors = {}
+    end
+
+    # State of the service
+    #
+    # In case the service is not active but socket, the socket state is considered
+    #
+    # @return [String]
+    def state
+      return socket.active_state if socket_active? && !service.active?
+
+      service.active_state
+    end
+
+    # Substate of the service
+    #
+    # In case the service is not active but socket, the socket substate is considered
+    #
+    # @return [String]
+    def substate
+      return socket.sub_state if socket_active? && !service.active?
+
+      service.sub_state
+    end
+
+    # Gets the current start_mode (as read from the system)
+    #
+    # @return [Symbol] :on_boot, :on_demand, :manual
+    def current_start_mode
+      @current_start_mode ||=
+        if service.enabled?
+          :on_boot
+        elsif socket && socket.enabled?
+          :on_demand
+        else
+          :manual
+        end
+    end
+
+    # Whether the service is currently active in the system
+    #
+    # @return [Boolean]
+    def current_active?
+      service.active? || socket_active?
+    end
+
+    # Returns the list of supported start modes
```

Comment:
+= "for this service (if a socket unit is available, :on_demand is supported, otherwise not)"

Reply:
Duh, it's kind of obvious, but someone might wonder why this is not a class constant instead.